htlcswitch/mock.go

```diff
@@ -225,6 +202,29 @@ func initSwitchWithDB(startingHeight uint32, db *channeldb.DB) (*Switch, error)
 	return New(cfg, startingHeight)
 }
 
+func initSwitchWithTempDB(t testing.TB, startingHeight uint32) (*Switch,
+	error) {
+
+	tempPath, err := ioutil.TempDir("", "switchdb")
```

Comment:
ioutil.TempDir is deprecated as of Go 1.17
use os.MkdirTemp instead

Reply:
On second thought might be better to be consistent with other usages and fix the deprecation in another pr